lsp/nls/src/requests/completion.rs

```diff
@@ -68,7 +68,10 @@ fn find_fields_from_contract(
 ) -> Option<Vec<Ident>> {
     let item = linearization.get_item(id)?;
     match &item.meta {
-        Some(meta_value) => Some(find_fields_from_meta_value(meta_value, path)),
+        Some(meta_value) => {
+            let result = find_fields_from_meta_value(meta_value, path);
+            (!result.is_empty()).then_some(result)
+        }
```

Comment:
I'm not sure to understand why this fixes the issue, but I imagine you tested the result. Is it because `find_fields_from_contracts` is called somewhere in a chain, and if it returns something, we keep the result, otherwise we try to find a completion by other means?

If this is case, why don't we aggregate all source of information (types, contracts, literal definition)? For example, if the contract is `{foo, ..}` but we know that the record is equal to `{foo = 1, bar = 2}`, does this mean that even after this PR, completion won't show `bar`?

Reply:
> Is it because find_fields_from_contracts is called somewhere in a chain, and if it returns something, we keep the result, otherwise we try to find a completion by other means?

Yes. 

I think aggregation is better. Thanks!